src/formBuilder/nftSelect.jsx

```diff
@@ -12,32 +12,65 @@ import {
   AspectRatio,
 } from '@chakra-ui/react';
 import deepEqual from 'deep-eql';
+import { useParams } from 'react-router-dom';
 
+import { useInjectedProvider } from '../contexts/InjectedProviderContext';
 import { useDao } from '../contexts/DaoContext';
 import { useOverlay } from '../contexts/OverlayContext';
 import FieldWrapper from './fieldWrapper';
+import GenericInput from './genericInput';
+import TokenInfoInput from './tokenInfoInput';
+import NftApproval from './nftApproval';
 import GenericModal from '../modals/genericModal';
+import TextBox from '../components/TextBox';
 import { filterUniqueNfts } from '../utils/nftVaults';
+import { fetchErc721s, fetchErc1155s } from '../utils/theGraph';
+import { hydrate721s, hydrate1155s } from '../utils/nftData';
 
 const NftSelect = props => {
-  const { label, localForm, htmlFor, name, localValues } = props;
-  const { register, setValue } = localForm;
-  const { setGenericModal } = useOverlay();
+  const { address } = useInjectedProvider();
   const { daoVaults } = useDao();
+  const { setGenericModal } = useOverlay();
+  const { daochain } = useParams();
+  const {
+    label,
+    localForm,
+    htmlFor,
+    name,
+    localValues,
+    source = 'dao',
+  } = props;
+  const { register, setValue } = localForm;
   const [nftData, setNftData] = useState();
   const [nfts, setNfts] = useState();
   const [selected, setSelected] = useState();
   const [collections, setCollections] = useState();
   const [filter, setFilter] = useState();
+  const [manualInput, setManualInput] = useState(false);
 
   useEffect(() => {
     register('tokenId');
     register('tokenBalance');
     register('raribleDescription');
     register('image');
     register('nftType');
+    register('tokenType');
     register('selectedMinion');
     register('selectedSafeAddress');
+
+    const getUserNfts = async () => {
```

Comment:
Is this happening on all proposal types or just the nft tribute?

Wondering how this is playing with the rarible forms and nft transfer forms

Reply:
oh does this control that:

````
if (source === 'user') {
```